extensions/amp-story/1.0/amp-story-tooltip.js

```diff
@@ -0,0 +1,401 @@
+     * element twice.
+     * @private {?Element} */
+    this.clickedElem_ = null;
+
+    this.storeService_.subscribe(StateProperty.TOOLTIP_ELEMENT, elem => {
+      if (!elem) {
+        return;
+      }
+
+      if (!this.isBuilt_) {
+        this.storyEl_.appendChild(this.build_());
+      }
+
+      if (elem != this.clickedElem_) {
+        this.clickedElem_ = elem;
+        this.attachTooltipToEl_(elem);
+      }
+      this.onTooltipStateUpdate_(true);
+    });
+  }
+
+  /**
+   * Builds the tooltip overlay and appends it to the provided story.
+   * @private
+   */
+  build_() {
+    this.isBuilt_ = true;
+
+    this.shadowRoot_ = this.win_.document.createElement('div');
+
+    this.tooltipOverlayEl_ =
+      dev().assert(this.buildTemplate_(this.win_.document));
+    createShadowRootWithStyle(this.shadowRoot_, this.tooltipOverlayEl_, CSS);
+
+    this.tooltipOverlayEl_
+        .addEventListener('click', event => this.onOutsideClickableEls_(event));
+
+    this.storeService_.subscribe(StateProperty.UI_STATE, isDesktop => {
+      this.onUIStateUpdate_(isDesktop);
+    }, true /** callToInitialize */);
+
+    this.storeService_.subscribe(StateProperty.CURRENT_PAGE_ID, () => {
+      // Hide active tooltip when page switch is triggered by keyboard or
+      // desktop buttons.
+      this.closeTooltip_();
+    });
+
+    return this.shadowRoot_;
+  }
+
+  /**
+   * Hides the tooltip layer.
+   * @private
+   */
+  closeTooltip_() {
+    this.storeService_.dispatch(Action.TOGGLE_TOOLTIP, null);
+    // Hide active tooltip when page switch is triggered by keyboard or
+    // desktop buttons.
+    this.onTooltipStateUpdate_(false);
+  }
+
+  /**
+   * Reacts to store updates related to tooltip active status.
+   * @param {boolean} tooltipIsActive
+   * @private
+   */
+  onTooltipStateUpdate_(tooltipIsActive) {
+    this.resources_.mutateElement(dev().assertElement(this.tooltipOverlayEl_),
+        () => {
+          this.tooltipOverlayEl_
+              .classList.toggle('i-amphtml-hidden', !tooltipIsActive);
+        });
+  }
+
+  /**
+   * Reacts to desktop state updates and hides navigation buttons since we
+   * already have in the desktop UI.
+   * @param {!UIType} uiState
+   * @private
+   */
+  onUIStateUpdate_(uiState) {
+    this.resources_.mutateElement(dev().assertElement(this.tooltipOverlayEl_),
+        () => {
+          uiState === UIType.DESKTOP ?
+            this.tooltipOverlayEl_.setAttribute('desktop', '') :
+            this.tooltipOverlayEl_.removeAttribute('desktop');
+        });
+  }
+
+  /**
+   * Builds tooltip and attaches it depending on the clicked element content and
+   * position.
+   * @param {!Element} clickedEl
+   * @private
+   */
+  attachTooltipToEl_(clickedEl) {
+    const {href} = parseUrlDeprecated(clickedEl.getAttribute('href'));
+
+    const domainName = getSourceOriginForElement(clickedEl, href);
+    const tooltipText = clickedEl.getAttribute('tooltipText') || domainName;
```

Comment:
HTML5 data attributes have to start with `data-`. Please rename these into `data-tooltip-text` and `data-tooltip-icon`